machine/v1/types_nutanixprovider.go

```diff
@@ -56,6 +56,25 @@ type NutanixMachineProviderConfig struct {
 	// +kubebuilder:validation:Required
 	SystemDiskSize resource.Quantity `json:"systemDiskSize"`
 
+	// bootType indicates the boot type (Legacy, UEFI or SecureBoot) the Machine's VM uses to boot.
+	// If this field is empty or omitted, the VM will use the default boot type "Legacy" to boot.
+	// "SecureBoot" depends on "UEFI" boot, i.e., enabling "SecureBoot" means that "UEFI" boot is also enabled.
+	// +kubebuilder:validation:Enum="";Legacy;UEFI;SecureBoot
+	// +optional
+	BootType NutanixBootType `json:"bootType,omitempty"`
```

Comment:
Since the UEFI and SecureBoot boot-type usually only used by Windows Machine vm, and it has the default value "Legacy", I think it is better to have the "omitempty".

Reply:
Removed